src/contract/HandlerBasedContract.ts

```diff
@@ -205,10 +207,16 @@ export class HandlerBasedContract<State> implements Contract<State> {
     if (!this.wallet) {
       throw new Error("Wallet not connected. Use 'connect' method first.");
     }
-    const { arweave } = this.smartweave;
 
-    const interactionTx = await this.createInteraction(input, tags, transfer, strict);
-    const response = await arweave.transactions.post(interactionTx);
+    const interactionTx = await this.createUnsignedInteraction(input, tags, transfer, strict);
+    await this.smartweave.arweave.transactions.sign(interactionTx, this.wallet);
+
+    return this.writeInteractionTx(interactionTx);
+  }
+
+  async writeInteractionTx(interactionTx: Transaction): Promise<string | null> {
```

Comment:
I believe all the public methods in this class should be reflected in the `Contract` interface - that this class is implementing

Reply:
done